contracts/misc/RewardSplitter.sol

```diff
@@ -134,10 +152,49 @@ contract RewardSplitter is IRewardSplitter, Initializable, OwnableUpgradeable, M
     uint256 rewards,
     address receiver
   ) external override returns (uint256 positionTicket) {
+    if (rewards == type(uint256).max) {
+      rewards = rewardsOf(msg.sender);
+    }
     _withdrawRewards(msg.sender, rewards);
     return IVaultEnterExit(vault).enterExitQueue(rewards, receiver);
   }
 
+  function enterExitQueueOnBehalf(uint256 rewards, address onBehalf) external {
+    if (!isClaimOnBehalfEnabled) return;
+
+    if (rewards == type(uint256).max) {
+      rewards = rewardsOf(onBehalf);
+    }
+    _withdrawRewards(onBehalf, rewards);
+    uint256 positionTicket = IVaultEnterExit(vault).enterExitQueue(rewards, address(this));
+    _exitPositions[positionTicket] = onBehalf;
+  }
+
+  function claimExitedAssetsOnBehalf(
+    uint256 positionTicket,
+    uint256 timestamp,
+    uint256 exitQueueIndex
+  ) external {
+    if (!isClaimOnBehalfEnabled) revert Errors.AccessDenied();
```

Comment:
Please remove this check, it should remain claimable even when claim on behalf is disabled.